### PR TITLE
Remove deprecated resources

### LINKED ISSUE #1741: compute floatingips
I am building a [migration script](https://github.com/gophercloud/gophercloud/pull/3085) for adapting codebases to Gophercloud v2.

In that context, I am wondering what to do with `compute/v2/extensions/floatingips`. That functionality is offered by Neutron at this point, and has been for quite a while. However, in `terraform-provider-openstack` `resourceComputeFloatingIPV2` and `resourceNetworkingFloatingIPV2` have a different API, and more specifically the legacy Compute version has the parameter `instance_id` that is not available in the Networking replacement.

So my question is: going forward, what do you want to do with `resourceComputeFloatingIPV2`, in the context of an upgrade to Gophercloud v2, which doesn't offer `floatingips` under `compute` any more?

### PR DESCRIPTION
**Pull request by kayrus**
Resolves #1741

The list of removed deprecated data sources:

* `openstack_blockstorage_snapshot_v2`
* `openstack_blockstorage_volume_v2`
* `openstack_fw_policy_v1`

The list of removed deprecated resources:

* `openstack_blockstorage_quotaset_v2`
* `openstack_blockstorage_volume_v1`
* `openstack_blockstorage_volume_v2`
* `openstack_blockstorage_volume_attach_v2`
* `openstack_compute_secgroup_v2` + fix examples in `networking_network_v2.md` doc
* `openstack_compute_floatingip_v2`
* `openstack_compute_floatingip_associate_v2` + fix examples in `compute_instance_v2.md` doc
* `openstack_fw_firewall_v1`
* `openstack_fw_policy_v1`
* `openstack_fw_rule_v1`
* `openstack_lb_member_v1`
* `openstack_lb_monitor_v1`
* `openstack_lb_pool_v1`
* `openstack_lb_vip_v1`

Fixed corresponding examples in the `examples` directory

### CODE REVIEW

**Review by nikParasyr — COMMENTED** [review]
Overall the PR is good and the guide also super useful.

My only problem is:
- I think this removal should come with a major release (3.0.0): Reasoning is that it is more aligned with semver versioning in general as well with [tf versioning best practices](https://developer.hashicorp.com/terraform/plugin/best-practices/versioning). The problem is that we will be cutting 2 major releases with 3 months but i prefer doing that instead of doing removals on a minor release.


So my proposed way forward would be:
- make sure that all the to-be-removed resources are marked as deprecated
- cut a 2.1.0 release now with any change that doesn't require  gophercloud v2
- work on a 3.0.0 release with the removal and the gophercloud v2 changes

File: docs/guides/upgrade-guide-version-2.1.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+---
+subcategory: ""
+layout: "openstack"
+page_title: "Terraform Openstack Provider Version 2.1 Upgrade Guide"
+description: |-
+  Terraform Openstack Provider Version 2.1 Upgrade Guide
+---
+
+Version 2.1.0 of the Openstack provider for Terraform is a major release and includes some changes that you will need to consider when upgrading. We intend this guide to help with that process.
```
[the remark this answers is not in the document]
Comment (kayrus) [reply docs/guides/upgrade-guide-version-2.1.md]: fixed

**Comment by kayrus** [comment]
> make sure that all the to-be-removed resources are marked as deprecated

they are

**Review by nikParasyr — APPROVED** [review]

**Comment by jaydub** [comment]
I'm attempting to figure out how to migrate from `openstack_compute_floatingip_associate_v2` to `openstack_networking_floatingip_associate_v2` and the experience has not been great. 

The key problem seems to be that the networking version wants a port id, but the instance's network state block lists the port as null. It's looking likely that I'll have to pull the mac address out of the networking block, feed that to a port data source and get the id from that, which is kinda janky.

It's possible that the port ought to have some value, but the rather old openstack neutron component I have no choice but to use is not doing the right thing — should I expect an id or something on that field?

(Overall, I would have preferred a deprecation-to-obsolete process that described the migration process on the deprecated resource at the point the provider starts emitting warnings, and for the resource docs to stick around, marked as obsolete for a major release, so people late to running `terraform init --upgrade` have an obvious place to look for guidance.)

**Comment by kayrus** [comment]
@jaydub have you followed the https://github.com/terraform-provider-openstack/terraform-provider-openstack/blob/9eea8814bfcf0dac90b4bf4c7d563ca971155fd1/docs/guides/upgrade-guide-version-3.md#compute-floating-ips-api-removal guide? You have to use the `openstack_networking_port_v2` data source.

**Pull request merged**